lib/membrane_element_aac/encoder.ex

```diff
@@ -0,0 +1,279 @@
+defmodule Membrane.Element.AAC.Encoder do
+  @moduledoc """
+  Element encoding raw audio into AAC format
+  """
+
+  use Membrane.Element.Base.Filter
+  use Bunch.Typespec
+  alias __MODULE__.Native
+  alias Membrane.Buffer
+  alias Membrane.Caps.Audio.Raw
+  alias Membrane.Caps.Matcher
+  alias Membrane.Event.EndOfStream
+
+  use Membrane.Log, tags: :membrane_element_aac
+
+  # AAC Constants
+  @aac_frame_size 1024
+  @sample_size 2
+
+  @default_channels 2
+  @default_sample_rate 44_100
+  # MPEG-4 AAC Low Complexity
+  @default_audio_object_type 2
+  @list_type allowed_channels :: [1, 2]
+  @list_type allowed_aots :: [2, 5, 29, 129, 132]
+  @list_type allowed_sample_rates :: [
+               96000,
+               88200,
+               64000,
+               48000,
+               44100,
+               32000,
+               24000,
+               22050,
+               16000,
+               12000,
+               11025,
+               8000
+             ]
+  @list_type allowed_bitrate_modes :: [0, 1, 2, 3, 4, 5]
+
+  @supported_caps {Raw,
+                   format: :s16le,
+                   channels: Matcher.one_of(@allowed_channels),
+                   sample_rate: Matcher.one_of(@allowed_sample_rates)}
+
+  def_options aot: [
```

Comment:
One more thing here - there should be a way to pass something more meaningful than a magic integer, like an atom, and the element should map it to integer appropriate for the underlying library